Add initial README.md for tests/ requirement files (#2577)
ansible/ansible-documentation: tests/README.md

# Descriptions of requirements files

The following table explains the purpose of the `.in` and `.txt` files in the `tests/` directory of this repository:

> 

Reviewer: I think one of the prior comments suggested adding a sentence about " that *.in is compiled into the corresponding *.txt with the general form of the uv pip compile command"

Author: Yeah, that was me. There was a "if that sounds useful" there too. I am unsure if that information is of value on this page. 

If it is, I am happy to add it in.

Reviewer: hey sorry it's taken me ages to get back to this comment. I think it's useful. I know I forget regularly how all this works :-)

| File            | Purpose                                                             | Pip Constraints file (when relevant) |
| ----            | -------                                                             | ------------------------------------ |

> |constraints.in   | Pins for sphinx and antsibull-docs, version bounds for known issues | --                                   |

Reviewer: ```suggestion
|constraints.in   | Pins/version bounds for sphinx and antsibull-docs  for known issues | --                                   |
```

|formatters.in    | List of Formatters required                                         | formatters.txt                       |

> |pip-compile.in   | Pin vesrion of `uv` for `pip-compile` nox session                   | pip-compile.txt                      |

Reviewer: ```suggestion
|pip-compile.in   | Requirements for `pip-compile` nox session                   | pip-compile.txt                      |
```

Reviewer: It currently doesn't have a bound or pinned version.

> |pip-compile.in   | Dependencies for `pip-compile` nox session                          | pip-compile.txt                      |

Reviewer 2: ```suggestion
|pip-compile.in   | Dependencies for `pip-compile` nox sessions                          | pip-compile.txt                      |
```

> |pip-compile.in   | Dependencies for `pip-compile` nox session                          | pip-compile.txt                      |

Reviewer 2: Maybe we could move this one down and group all the nox sessions?

> |pr_labeler.in    | Dependencies for pr_labeler                                         | pr_labeler.txt                       |

Reviewer 2: ```suggestion
|pr_labeler.in    | Dependencies for the labeler workflow                                         | pr_labeler.txt                       |
```

> |requirements.in  | Dependencies file for Docs Build                                    | requirements.txt                     |

Reviewer 2: ```suggestion
|requirements.in  | Dependencies file for Sphinx doc builds                                    | requirements.txt                     |
```

> |spelling.in      | Dependencies for `spelling` nox session                             | spelling.txt                         |

Reviewer 2: ```suggestion
|spelling.in      | Dependencies for the `spelling` nox session                             | spelling.txt                         |
```

> |static.in        | Dependencies for `static` nox session                               | static.txt                           |

Reviewer 2: ```suggestion
|static.in        | Dependencies for the `static` nox session                               | static.txt                           |
```

> |tag.in           | Dependencies for `tag` nox session                                  | tag.txt                              |

Reviewer 2: ```suggestion
|tag.in           | Dependencies for the `tag` nox session                                  | tag.txt                              |
```

> |typing.in        | Dependencies for `typing` nox session                               | typing.txt                           |

Reviewer 2: ```suggestion
|typing.in        | Dependencies for the `typing` nox session                               | typing.txt                           |
```

> # Requirements Files, and their uses

Reviewer 2: ```suggestion
# Descriptions of requirements files

The following table explains the purpose of the `.in` and `.txt` files in the `tests/` directory of this repository:

```

| File            | Purpose                                                                                            | 
| ----            | -------                                                                                            |

> |constraints.in   | Pins for the stable, tested versions of sphinx and antsibull-docs that production builds rely upon |

Reviewer 3: Technically, not just pins. Actually, pins should be avoided in there mostly. Only specific version exclusions and rare top boundaries should go here. What's known to be broken (temporarily).

Author: I pulled the language right out of the file. Is there a better way to phrase it? 

Something like "Pins, upper or lower version boundaries for known issues"

|formatters.in    | List of Formatters required                                                                        |
|pip-compile.in   | Pin vesrion of `uv` for `pip-compile` nox session                                                  |
|pr_labeler.in    | Requirements for pr_labeler                                                                        |
|requirements.in  | Requirements file for Docs Build                                                                   |
|spelling.in      | Requirements for `spelling` nox session                                                            |
|static.in        | Requirements for `static` nox session                                                              |
|tag.in           | Requirements for `tag` nox session                                                                 |
|typing.in        | Requirements for `typing` nox session                                                              |
|formatters.txt   | Output of `uv pip compile --universal --output-file tests/formatters.txt -r tests/formatters.in`   |

> |pip-compile.txt  | Output of `uv pip compile --universal --output-file tests/pip-compile.txt tests/pip-compile.in`    |

Reviewer 3: txt files are pins (pip constraints really)

Reviewer 3: Maybe add a separate column for txt files mapped to in? It's mostly 1:1.

Author: > Maybe add a separate column for txt files mapped to in? It's mostly 1:1.

As in, move the `*.txt` files to another column on the lines with `*.in` files, as opposed to their own lines? 

That sounds totally reasonable. Then maybe (if its useful to know?) just a singular blurb that `*.in` is compiled into the corresponding `*.txt` with the general form of the `uv pip compile` command.
